README.md
# FactorioMods
## Some or all of the mods I've created/modified for Factorio.
### note, these zips are mods from others that I play, they are from the devs itself, not modified!
### U should check then out in the official factorio mods page.
### I'm just uploading those here in case I lost the names/zips ( happened b4 )
### my mods are the uncompressed ones.



ZFixes/changelog.txt
---------------------------------------------------------------------------------------------------
Version: 1.3.4
Date: 2023.11.11
  Info:
    - Trying to improve the code (always).
    - Auto unlock tech: Optics.
    - Steam furnance now correctly depends on bobspower.
    - Lots of corretions to item names and description.
    - Option to toggle friendly_fire.
    - Mod description improvments.
---------------------------------------------------------------------------------------------------
Version: 1.3.3
Date: 2022.10.08
  Info:
    - Finally, the code is fully identated.
    - Fixed misplaced "Ghost Time To Live" code, was a duplicate.
    - Fixed misplaced code line that was doing nothing but taking space.
    - Fixed "Steam Furnace" always unlocked, even without steel-furnace tech avaliable.
    - Removed K2_WirelessCharger support.
    - Added a Mini 2x2 Radar Thanks to Endus322.
    - Removed HeatGlow stuff.
    - Nerfed my default Character speed (1.5 > 1.2)
---------------------------------------------------------------------------------------------------
Version: 1.3.2
Date: 2021.10.30
  Info:
    - Added option to start with Antron and Spider remote.
---------------------------------------------------------------------------------------------------
Version: 1.3.1
Date: 2021.10.12
  Info:
    - Added option to start with Antron.
    - Compressed images even more to reduce mod size.
---------------------------------------------------------------------------------------------------
Version: 1.3.0
Date: 2021.10.10
  Info:
    - Fixed some spelling ( Uh, I gotta improve my English... ).
    - Reduced Manual Mining speed from 1.5 to 1.3.
    - Updated HeatGlow.
    - Updated Roboports Radius.
    - Tweaked Steam-Furnace.
    - K2_WirelessCharger Energy Absorber is now 100 copper cables more expensive.
    - Little buff on ShockTurret damage ( 4 to 6 ).
    - Landfill is now 15 stone.
    - Compressed Images and Code to reduce mod size.
    - Changed Chests Recipes.
    - Set Logistic Delivery Icon to a smaller one.
---------------------------------------------------------------------------------------------------
Version: 1.2.9
Date: 2021.03.13
  Info:
    - Option to start with a car.
    - Fixed some spelling.
    - Even less red.
    - Character running speed is now configurable ( suggestion from TolgahanTR ).
    - Mod now advise on runtime settings are changed/updated.
---------------------------------------------------------------------------------------------------
Version: 1.2.8
Date: 2021.03.10
  Info:
    - Reduced LabSpeed ( From cheat option ).
    - Reduced LabProductivity ( From cheat option ).
    - Now you start a new map with some free techs ( if SuperQOL is enabled )( see mod description ).
    - Tweaked some code.
    - Respawn time is now configurable.
    - Fixed in game changelog.
---------------------------------------------------------------------------------------------------
Version: 1.2.7
Date: 2021.03.07
  Info:
    - Full dark nights is now enable by default.
    - Made landfill even more cheap ( from 15 to 13 ).
    - Reduced LabSpeed ( From cheat option ).
    - Reduced LabProductivity ( From cheat option ).
    - Entity ghosts enabled by default.
---------------------------------------------------------------------------------------------------
Version: 1.2.6
Date: 2021.02.28
  Info:
    - Boilers will quickstart with 10 water units.
---------------------------------------------------------------------------------------------------
Version: 1.2.5
Date: 2021.02.25
  Info:
    - Improved contrast of Logistic Chests.
    - Improved heat-pipes glow a little bit more.
    - Fixed Small-Power-Pole wire reach and supply distance.
    - Reduced build and reach distance from 300 to 200.
    - Changed more settings of K2_WirelessCharger
    - Removed no-lakes changes.
    - Now you can place wood as floor Thanks to zanven.
    - Improved HeatGlow for heat-pipes, reactor, and heat-exchanger Thanks to max2344.
---------------------------------------------------------------------------------------------------
Version: 1.2.4
Date: 2021.02.19
  Info:
    - Added option to increase all solar panels energy production by 4 times and increase accumulators by 2 times.
    - Fixed missing title for Linked-Chest technology.
    - Reduced some code.
    - Removed BigTruck changes.
    - Removed Wood-Walls changes.
    - Added option to make oil finite.
    - Changed settings of ShockTurret.
    - Improved Sniper Range ( Bob's Warfare ) from 30 to 40.
    - Improved heat-pipes glow Thanks to max2344.
    - Added a better icon to Linked-Chest technology.
---------------------------------------------------------------------------------------------------
Version: 1.2.3
Date: 2021.02.10
  Info:
    - Quick fix to Linked Chest 'signal' now you can select it as icon for blueprint books and etc. Thanks to @Doggo on Discord.
    - Linked-Chest inventory size is now 19 slots.
    - Linked-Chest is now in logistics group.
    - Added option to change corpse duration.
    - Fixed some spelling.
    - Fixed setting names ( forgot to add ;-; ).
---------------------------------------------------------------------------------------------------
Version: 1.2.2
Date: 2021.02.09
  Info:
    - Changed some settings of K2_WirelessCharging.
    - Changed some settings of Wood-Walls.
    - Added option to enable full dark nights.
    - Added option to reduce the red flash intensity when you get dammaged.
    - Changed some settings of bigtruck ( rebalance ).
---------------------------------------------------------------------------------------------------
Version: 1.2.1
Date: 2021.02.05
  Info:
    - Beacons can now utilize all modules.
    - Beacons have now a little bit bigger supply area.
    - Beacons have now a 4 module slots.

ZElectricVehicles/data.lua
local ICONPATH = "__ZElectricVehicles__/graphics/icons/"

data:extend{
  {
    type = "battery-equipment",
    name = "zelectric-transformer",
    categories = {"armor"},
    energy_source = {
      type = "electric",
      buffer_capacity = "15MJ",
      usage_priority = "secondary-input"
    },
    shape = {
      type = "full",
      width = 3,
      height = 3,
    },
    sprite = {
      filename = ICONPATH .. "transformer.png",
      width = 192,
      height = 192,
      scale = 0.5,
    },
  },
  {
    type = "item",
    name = "zelectric-transformer",
    icon = ICONPATH .. "transformer-icon.png",
    icon_size = 64,
    subgroup = "equipment",
    order = "b[battery]-d[electric-transformer]",
    stack_size = 5,
    placed_as_equipment_result = "zelectric-transformer"
  },
  {
    type = "recipe",
    name = "zelectric-transformer",
    category = "crafting",
        ingredients = {
            {"iron-plate", 50},
            {"copper-cable", 300},
            {"plastic-bar", 5},
            {"electric-engine-unit", 10},
        },
        results = {
          {type = "item", name = "zelectric-transformer", amount = 1}},
        enabled = "false",
    energy_required = 2,
    subgroup = "equipment",
  },

  {
    type = "technology",
    name = "zelectric-transformer",
    icon = ICONPATH .. "transformer-tech.png",
    icon_size = 256,
    prerequisites = {"advanced-electronics-2", "battery-equipment", "electric-engine"},
    effects =
    {
      {
        type = "unlock-recipe",
        recipe = "zelectric-transformer"
      },
    },
    unit = {
      count = 200,
      ingredients = {
        {"automation-science-pack", 1},
        {"logistic-science-pack", 1},
        {"chemical-science-pack", 1}
      },
      time = 30
    },
    order = "g-i-a"
  },

  {
    type = "item",
    name = "zelectric-transformer-power",
    icon = ICONPATH .. "fuel.png",
    icon_size = 64,
    fuel_value = "15MJ",
    fuel_category = "chemical",
    fuel_acceleration_multiplier = settings.startup["ZElectricVehicles-Acceleration"].value,
    fuel_top_speed_multiplier = settings.startup["ZElectricVehicles-TopSpeed"].value,
    stack_size = 1,
    flags = {"hide-from-fuel-tooltip"},
  },
}


